nest-cli/src/commands/operations/install.rs

```diff
@@ -0,0 +1,54 @@
+use failure::{format_err, Error, ResultExt};
+use indicatif::{ProgressBar, ProgressStyle};
+use libnest::config::Config;
+use libnest::transaction::InstallTransaction;
+
+use super::download::Download;
+
+pub fn install_package(config: &Config, trans: &mut InstallTransaction) -> Result<(), Error> {
```

Comment:
The way you handle `install` doesn't separate downloads from applying the transaction.

When you want do install A, B and C, you will do:

```
Download A
Install A
Download B
Install B
Download C
Install C
```

This is a bit dangerous because if the whole operation is canceled mid way, we will have some packages installed while others are missing. You can reduce this effect (but not completely make it disappear) by first downloading all needed packages and then installing them.

```
Download A
Download B
Download C
Install A
Install B
Install C
```

Such strategy can also achieve parallel downloads more easily and in a safer manner:

```
Download A, B and C using different threads
=== Wait for all downloads to finish ===
Install A
Install B
Install C
```

Installations aren't done concurrently because it's unsafe to do so, and may result in an unstable system.

Reply:
At the moment, it is indeed a single block, and it will probably remain that way until I add the `merge` command (which also uses the install operation). After that only will I split it / think about parallelization opportunities